pkg/operator/api/informer/informer_test.go

```diff
@@ -75,6 +75,11 @@ func Test_WatchUpdates(t *testing.T) {
 			},
 			operator.ResourceEventType_UPDATED,
 		)
+
+		require.Eventually(t, func() bool {
+			return i.batchID == 1
+		}, 5*time.Second, 100*time.Millisecond)
```

Comment:
We want to print out the values if it does fail. This also has a data race so we need to change `i.batchID` to be a `atomic.Uint32.

```suggestion
		require.EventuallyWithT(t, func(c *assert.CollectT) {
			assert.Equal(t, 1, i.batchID)
		}, 5*time.Second, 100*time.Millisecond)
```